walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,66 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<String> attendees = request.getAttendees();
+    Collection<TimeRange> answers = new ArrayList<TimeRange>();
+
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+        return answers;
+    }
+
+    List<TimeRange> relatedEvents = getRelatedEvents(events, attendees);
+
+    if (relatedEvents.isEmpty()) {
+        answers.add(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TimeRange.END_OF_DAY, true));
+        return answers;
+    }
+
+    Collections.sort(relatedEvents, TimeRange.ORDER_BY_START);
+
+    int pointerA = 0, pointerB = 0;
+
+    if (TimeRange.START_OF_DAY + request.getDuration() <= relatedEvents.get(0).start()) {
```

Comment:
how about this part from 42-61?

Reply:
It's up to you, but I think it's not that necessary because this is the main "meat" of the function

Reply:
icic